source/common/upstream/load_balancer_impl.cc

```diff
@@ -1299,25 +1299,88 @@ HostConstSharedPtr LeastRequestLoadBalancer::unweightedHostPick(const HostVector
                                                                 const HostsSource&) {
   HostSharedPtr candidate_host = nullptr;
 
+  switch (selection_method_) {
+    case envoy::extensions::load_balancing_policies::least_request::v3::LeastRequest_SelectionMethod_FULL_SCAN:
+      candidate_host = unweightedHostPickFullScan(hosts_to_use);
+      break;
+    case envoy::extensions::load_balancing_policies::least_request::v3::LeastRequest_SelectionMethod_N_CHOICES:
+    default:
+      candidate_host = unweightedHostPickNChoices(hosts_to_use);
```

Comment:
I think these enum names are the mangled top-level names. See https://protobuf.dev/reference/cpp/cpp-generated/#enum, which is referenced from our style guide.

Fall-through cases need to be expliticly marked, but in general we use something like `IS_ENVOY_BUG("unknown selection method")` for unknown enum values. In the case that a num enum value is added in the future, doing that makes it easier to find the places that it needs to be supported.

Reply:
(The validation clause on the field should prevent an older Envoy from accepting a config with a newer version of the API's enum values. The config should be rejected long before it gets here.)